frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -191,39 +191,49 @@ local footerTextGeneratorMap = {
                                           footer.ui.pagemap:getLastPageLabel(true))
             end
-            return ("%d / %d"):format(footer.pageno, footer.pages)
+            if footer.ui.document:hasFlows() then
+                -- i.e., if we are hiding non-linear fragments and there's anything to hide,
+                if footer.flow == 0 then
+                    return ("%d // %d"):format(footer.pageno, footer.pages)
+                else
+                    return ("[%d / %d]%d"):format(footer.pageno, footer.pages, footer.flow)
+                end
+            else
+                return ("%d / %d"):format(footer.pageno, footer.pages)
+            end
         elseif footer.position then
             return ("%d / %d"):format(footer.position, footer.doc_height)
         end
     end,
     pages_left = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].pages_left
-        local left = footer.ui.toc:getChapterPagesLeft(footer.pageno)
+        local left = footer.ui.toc:getChapterPagesLeft(footer.ui:getCurrentPage())
         return prefix .. " " .. (left and left or footer.pages - footer.pageno)
     end,
     percentage = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].percentage
         local digits = footer.settings.progress_pct_format or "0"
-        local string_percentage
-        if not prefix then
-            string_percentage = "%." .. digits .. "f%%"
-        else
-            string_percentage = prefix .. " %." .. digits .. "f%%"
+        local string_percentage = "%." .. digits .. "f%%"
+        if footer.flow ~= 0 then
+            string_percentage = "[" .. string_percentage .. "]"
+        end
+        if prefix then
+            string_percentage = prefix .. " " .. string_percentage
         end
         return string_percentage:format(footer.progress_bar.percentage * 100)
     end,
     book_time_to_read = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].book_time_to_read
-        local current_page = footer.ui:getCurrentPage()
-        return footer:getDataFromStatistics(prefix .. " ", footer.pages - current_page)
+        local left = footer.ui.document:getTotalPagesLeft(footer.ui:getCurrentPage())
+        return footer:getDataFromStatistics(prefix .. " ", left)
     end,
     chapter_time_to_read = function(footer)
         local symbol_type = footer.settings.item_prefix or "icons"
         local prefix = symbol_prefix[symbol_type].chapter_time_to_read
-        local left = footer.ui.toc:getChapterPagesLeft(footer.pageno)
+        local left = footer.ui.toc:getChapterPagesLeft(footer.ui:getCurrentPage())
         return footer:getDataFromStatistics(
             prefix .. " ", (left and left or footer.pages - footer.pageno))
```

Comment:
Not affected by this PR, but is this construction needed? Wouldn't `(left or footer.pages - footer.pageno)` be exactly the same?

Reply:
Looks to me that indeed, it's exactly the same (unless I miss something with Lua pecularities :)

Reply:
If there are no ticks at all (i.e., no chapter breaks), `ReaderToc:getNextChapter` will return `nil`, and `ReaderToc:getChapterPagesLeft` will return it as-is.

Reply:
That explains the need for `left or ...`, but not for `left and left`? If `left` is `nil`, `left or "foo"` is `"foo"`, and `left and left or "foo"` is `"foo"` as well... at least as far as I can see.

Reply:
Oh, yeah, I totally skipped over the fact that there was an `or` after :D.